R/summarize_weighted_production.R

```diff
@@ -53,37 +53,97 @@
 #'
 #' summarize_weighted_percent_change(master, use_credit_limit = TRUE)
 summarize_weighted_production <- function(data, ..., use_credit_limit = FALSE) {
+  summarize_weighted_production_(data, ..., use_credit_limit = use_credit_limit, add_targets = FALSE)
+}
+
+summarize_weighted_production_ <- function(data, ..., use_credit_limit = FALSE, add_targets = FALSE) {
```

Comment:
I have added this new internal function `summarize_weighted_production_` which contains the argument `add_targets`. I then call `summarize_weighted_production` with default values for that internal function (this is to preserve the existing arguments of `summarize_weighted_production`). 

@maurolepore let me know what you think of this solution as a way to avoid WET code